File: app/src/main/java/mega/privacy/android/app/receivers/ChargeEventReceiver.java
package mega.privacy.android.app.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import static mega.privacy.android.app.utils.JobUtil.*;
import static mega.privacy.android.app.utils.LogUtil.*;


public class ChargeEventReceiver extends BroadcastReceiver {
    
    @Override
    public void onReceive(final Context context,Intent intent) {
        logDebug("ChargeEventReceiver");
        startCameraUploadService(context);
    }
}

File: app/src/main/java/mega/privacy/android/app/receivers/ChargeEventReceiver.kt
package mega.privacy.android.app.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.Intent.ACTION_POWER_CONNECTED
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch
import mega.privacy.android.domain.qualifier.ApplicationScope
import mega.privacy.android.domain.usecase.workers.ScheduleCameraUploadUseCase
import mega.privacy.android.domain.usecase.workers.StartCameraUploadUseCase
import timber.log.Timber
import javax.inject.Inject

/**
 * Broadcast receiver triggered when an external power has been connected to the device
 */
@AndroidEntryPoint
class ChargeEventReceiver : BroadcastReceiver() {

    /**
     * Application scope to run coroutines
     */
    @ApplicationScope
    @Inject
    lateinit var applicationScope: CoroutineScope

    /**
     *  Start camera upload
     */
    @Inject
    lateinit var startCameraUploadUseCase: StartCameraUploadUseCase

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action.equals(ACTION_POWER_CONNECTED)) {
            Timber.d("ChargeEventReceiver")
            applicationScope.launch { startCameraUploadUseCase() }
        }
    }
}
